WeScan/ImageScannerController.swift

```diff
@@ -129,25 +129,72 @@ public final class ImageScannerController: UINavigationController {
             self.blackFlashView.isHidden = true
         }
     }
+}
+
+/// Data structure containing information about a scan, including both the image and an optional PDF.
+public struct ImageScannerScan {
+    public enum ImageScannerError: Error {
+        case failedToGeneratePDF
+    }
+    
+    public var image: UIImage
+    
+    public func generatePDFData(completion: @escaping (Result<Data, ImageScannerError>) -> Void) {
+        DispatchQueue.global(qos: .userInteractive).async {
```

Comment:
To prevent retain cycles:

```suggestion
        DispatchQueue.global(qos: .userInteractive).async { [weak self] in
```

Reply:
I don't think this is possible/needed as `ImageScannerScan` is a value type/struct? Let me know if I'm mistaken:
`'weak' may only be applied to class and class-bound protocol types, not 'ImageScannerScan'`

https://stackoverflow.com/questions/38841127/why-can-the-keyword-weak-only-be-applied-to-class-and-class-bound-protocol-typ/44483737

Reply:
You're completely right, Xcode will even warn you for that. I missed that it was a struct! 👌